pkg/vcs/linux_configs.go

```diff
@@ -83,3 +89,60 @@ func linuxConfigsForTags(cf *kconfig.ConfigFile, tags map[string]bool) {
 		}
 	}
 }
+
+// linuxConfigsForType removes Linux kernel config options that are not necessary for the specific
+// report type.
+func linuxConfigsForType(cf *kconfig.ConfigFile, typ dashapi.CrashType, dt debugtracer.DebugTracer) {
+	if typ == dashapi.Warning {
+		typ = dashapi.Bug
+	}
+	keep := map[dashapi.CrashType]func(){
+		dashapi.Hang: func() {
+			cf.Unset("RCU_STALL_COMMON")
+			cf.Unset("LOCKUP_DETECTOR")
+			cf.Unset("DETECT_HUNG_TASK")
+			val := cf.Value("CMDLINE")
+			if val == "" {
+				val = `""`
+			}
+			pos := strings.LastIndexByte(val, '"')
+			if pos >= 0 {
+				cf.Set("CMDLINE",
+					val[:pos]+" rcupdate.rcu_cpu_stall_suppress=1"+val[pos:])
```

Comment:
Maybe we can set RCU_CPU_STALL_TIMEOUT=1000 and RCU_EXP_CPU_STALL_TIMEOUT=0 to keep code simpler?

Reply:
@dvyukov  These don't seem to fully prevent `INFO: rcu detected stall in` crashes.